api/all/src/main/java/io/opentelemetry/api/baggage/propagation/W3CBaggagePropagator.java

```diff
@@ -48,20 +50,29 @@ private W3CBaggagePropagator() {}
     if (baggage.isEmpty()) {
       return;
     }
+    StringBuilder headerContent = baggageToString(baggage);
+
+    if (headerContent.length() > 0) {
```

Comment:
I think better to do this in `baggageToString` and return a `String` instead of builder